pkg/cloudprovider/cloudprovider.go

```diff
@@ -124,14 +130,14 @@ func (c *CloudProvider) Create(ctx context.Context, nodeClaim *karpv1.NodeClaim)
 
 	instanceTypes, err := c.resolveInstanceTypes(ctx, nodeClaim, nodeClass)
 	if err != nil {
-		return nil, fmt.Errorf("resolving instance types, %w", err)
+		return nil, cloudprovider.NewCreateError(fmt.Errorf("resolving instance types, %s", truncateMessage(err.Error())), InstanceTypeResolutionFailedReason, err.Error())
```

Comment:
Here and below: Please truncate the message (last argument to NewCreateError) and not the error (first argument). The actual error passed is not even used when CreatedError is returned ([evidence](https://github.com/kubernetes-sigs/karpenter/blob/7bcbabf822b2ed2a1eab8750763f7e0c405874f4/pkg/controllers/nodeclaim/lifecycle/launch.go#L105-L106))

```suggestion
		return nil, cloudprovider.NewCreateError(fmt.Errorf("resolving instance types, %s", err.Error()), InstanceTypeResolutionFailedReason, truncateMessage(err.Error()))
```

Reply:
Fixed all 3 to be truncating the last argument. (Except the first one, the first one doesn't really have a full error)